### PR TITLE
Add css variables to components

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

(*Link to bug/feature from [Issues](../issues) and how you resolve it*)

Add css variable to give user of the following omni input components the ability to style the auto select hover color of the internal input element 

omni-email-field
omni-number-field
omni-pin-field
omni-search-field
omni-text-field

Expose a new css variable on the Modal component to give users the ability to apply a min width to the component

Added blur on enter for Search field component

Testing was done via a alpha

### All Submissions:

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/modal/Modal.ts [hunk 1]
```diff
@@ -254,6 +255,7 @@ export class Modal extends OmniElement {
                     background: transparent;
                     box-shadow: var(--omni-modal-container-box-shadow);
                     max-width: var(--omni-modal-max-width,100%);
+                    min-width: var(--omni-modal-min-width, auto);
```
Comment (BOTLANNER) [inline src/modal/Modal.ts:258]: Is this backward compatible? Shouldn't it be unset instead of auto?

File: src/search-field/SearchField.ts [hunk 2]
```diff
@@ -85,6 +87,12 @@ export class SearchField extends OmniFormElement {
         this.value = input?.value;
     }
 
+    _blurOnEnter(e: KeyboardEvent) {
```
Comment (BOTLANNER) [inline src/search-field/SearchField.ts:90]: Is this used?

File: src/modal/Modal.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/modal/Modal.ts:258]: According to MDN reference documentation the default for this is auto welcome to check the following link for reference. 

https://developer.mozilla.org/en-US/docs/Web/CSS/min-width

File: src/search-field/SearchField.ts (hunk 2 above)
Comment (chromaticWaster) [reply src/search-field/SearchField.ts:93]: Added event listener to call this function accordingly and tested locally.

**Comment by chromaticWaster** [comment]
Note there is currently a issue on the docs site where the Icon component stories do not render as expected hence those specific tests failing the following ticket has been logged as a issue to address this.

In the case of the Select component a issue should be logged for that this results in screenshot comparisons being inconsistent and usually results in the tests for this component failing.

https://github.com/capitec/omni-components/issues/219

File: src/modal/Modal.ts (hunk 1 above)
Comment (stefan505) [reply src/modal/Modal.ts:258]: I suspect `unset` would equate to `auto`. Happy either way.

**Review by stefan505 — APPROVED** [review]

**Comment by chromaticWaster** [comment]
Merging this the current screenshot comparison for some of the input fields are out by 1 pixel.
Cases for failures for Icon and Select component is logged as an issue that needs to be addressed in future

**Pull request merged**